src/util/metadata/side_metadata/global.rs

```diff
@@ -182,64 +182,92 @@ impl SideMetadataSpec {
         }
     }
 
-    /// Bulk-zero a specific metadata for a chunk. Note that this method is more sophisiticated than a simple memset, especially in the following
-    /// cases:
-    /// * the metadata for the range includes partial bytes (a few bits in the same byte).
-    /// * for 32 bits local side metadata, the side metadata is stored in discontiguous chunks, we will have to bulk zero for each chunk's side metadata.
-    ///
-    /// # Arguments
-    ///
-    /// * `start`: The starting address of a memory region. The side metadata starting from this data address will be zeroed.
-    /// * `size`: The size of the memory region.
-    ///
-    pub fn bzero_metadata(&self, start: Address, size: usize) {
-        #[cfg(feature = "extreme_assertions")]
-        let _lock = sanity::SANITY_LOCK.lock().unwrap();
+    /// This method is used for bulk setting side metadata for a data address range. As we cannot guarantee
+    /// that the data address range can be mapped to whole metadata bytes, we have to deal with cases that
+    /// we need to mask and set certain bits in a metadata byte.
+    /// The end address and the end bit are exclusive.
+    pub(super) fn set_meta_bits(
```

Comment:
Just like you parameterised `bulk_update_metadata` with the `update_meta_bits` parameter, you can merge `set_meta_bits` and `zero_meta_bits` and parameterise the setting/clearing operation with a parameter, too.

Reply:
Done. I added `update_meta_bits` which takes two function arguments, `update_bytes` and `update_bits`. `set_meta_bits` and `zero_meta_bits` both use `update_meta_bits` with different update bytes/bits closures.